uitools/widgets/Esri/ArcGISRuntime/Toolkit/BasemapGallery.h

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ *  Copyright 2012-2021 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#ifndef ESRI_ARCGISRUNTIME_TOOLKIT_BASEMAPGALLERY_H
+#define ESRI_ARCGISRUNTIME_TOOLKIT_BASEMAPGALLERY_H
+
+#include "Basemap.h"
+#include "BasemapGalleryController.h"
+
+#include <QFrame>
+namespace Ui {
+  class BasemapGallery;
+}
+
+namespace Esri {
+namespace ArcGISRuntime {
+namespace Toolkit {
+  class BasemapGalleryController;
+
+  class BasemapGallery : public QFrame
+  {
+    Q_OBJECT
+
+  public:
+    Q_INVOKABLE explicit BasemapGallery(QWidget* parent = nullptr);
+    Q_INVOKABLE explicit BasemapGallery(GeoModel* basemap, QWidget* parent = nullptr);
+
+    ~BasemapGallery() override;
+
+    BasemapGalleryController* controller() const;
+    void setGeoModel(GeoModel* geomodel);
+    GeoModel* getCurrentGeoModel();
+
+    //signals
+    void clickedItem(const QModelIndex& index);
```

Comment:
💭  I'm actually surprised this works! This isn't a `signal` it is a `slot`, and normally slots must be declared in a `public slots:`/`private slots:` scope.

I have another comment on removing this method alltogether, so this is more an observation.

Reply:
https://stackoverflow.com/questions/9147636/qt-private-slots-what-is-this#comment75141153_9147690

Reply:
👍 Gotcha, it works but not with introspection.